src/tree_row_reference.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright 2013-2016, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use glib;
+use glib::translate::*;
+use glib::object::IsA;
+use libc::c_int;
+use TreeIter;
+use TreePath;
+use TreeRowReference;
+
+impl TreeRowReference {
+
+    pub fn reordered<T: IsA<glib::Object>>(proxy: &T, path: &TreePath, iter: Option<&TreeIter>, new_order: &[u32]) {
+        assert_initialized_main_thread!();
+        assert!(iter.is_some() || path.get_depth()==0, "If 'iter' is None, 'path' must point to the root.");
```

Comment:
I think we don't have access to the TreeModel (proxy can be anything) which we would need for figuring out the size. But, [according to the GTK docs](https://developer.gnome.org/gtk3/stable/GtkTreeModel.html#gtk-tree-row-reference-new-proxy), the whole proxy thing is only public for rare monitoring reasons and TreeRowReference::reordered is then directly connected to TreeModel's reordered signal. Hence, in this case all arguments are valid.